api/src/context.rs

```diff
@@ -900,17 +902,30 @@ impl Context {
         if !config.enabled {
             return Ok(self.default_gas_estimation(min_gas_unit_price));
         }
+        if let Some(static_override) = &config.static_override {
+            return Ok(GasEstimation {
+                deprioritized_gas_estimate: Some(static_override.low),
+                gas_estimate: static_override.market,
+                prioritized_gas_estimate: Some(static_override.aggressive),
+            });
+        }
 
         let epoch = ledger_info.epoch.0;
 
         // 0. (0) Return cached result if it exists
-        if let Some(cached_gas_estimation) = self.cached_gas_estimation(epoch) {
+        let cache = self.gas_estimation_cache.read().unwrap();
+        if let Some(cached_gas_estimation) = self.cached_gas_estimation(&cache, epoch) {
             return Ok(cached_gas_estimation);
         }
+        drop(cache);
 
         // 0. (1) Write lock and prepare cache
         let mut cache = self.gas_estimation_cache.write().unwrap();
-        // TODO: retry cached result after acquiring write lock
+        // Retry cached result after acquiring write lock
```

Comment:
Will anything bad happen if you don't fix this race? 😄 Just curious.

Reply:
Nice, I was wondering if the issue was just recomputation or if there was a more serious correctness concern, but this makes sense 😄